File: C++/프로그래머스/1/161989. 덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    vector<bool> visited(section.size());
    int answer = 0;
    
    int maxLen;
    for(int i = 0; i < section.size(); i++) {
        if(!visited[i]) { //칠 되어있지 않을 경우
            visited[i] = true;
            answer++;
            maxLen = section[i] + m;
            while(i + 1 < section.size() && section[i + 1] < maxLen) {
                visited[++i] = true;
                
            }
        }
    }
    
    return answer;
}

File: Python3/프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    visited = [False] * len(section)
    
    answer = 0
    
    for i, (v, s) in enumerate(zip(visited, section)):
        if not v:
            answer += 1
            while i < len(section) and section[i] < s + m:
                visited[i] = True
                i += 1
            # print("section:", section)
            # print("visited:", visited)
            
    
    return answer